src/main/java/io/hhplus/tdd/point/UserPoint.java

```diff
@@ -9,4 +9,28 @@ public record UserPoint(
     public static UserPoint empty(long id) {
         return new UserPoint(id, 0, System.currentTimeMillis());
     }
+
+    public UserPoint charge(long point) {
+        if (point < 1) {
+            throw new IllegalArgumentException("최소 충전 금액은 1포인트 입니다");
+        }
+        long maxPoint = 1000L;
+        long currentPoint = this.point() + point;
+        if (currentPoint > maxPoint) {
+            throw new IllegalArgumentException("최대 잔고 제한을 초과했습니다. 현재 포인트: " + currentPoint);
+        }
+        return new UserPoint(this.id(), currentPoint, System.currentTimeMillis());
+    }
+
+    public UserPoint use(long point) {
+        if (point < 1) {
+            throw new IllegalArgumentException("최소 사용 금액은 1포인트 입니다");
+        }
+        long minPoint = 0L;
+        long currentPoint = this.point() - point;
+        if (currentPoint < minPoint) {
```

Comment:
```suggestion
        if (this.point < amount) {
```
이렇게 정리되면 더 가독성에 좋을 거 같아요!

Reply:
if (currentPoint < minPoint) , if (this.point < amount) 이 둘이 동일한 비교문이 아니지 않나요..?

Reply:
long minPoint와 같이 정책적인 부분은 메서드에서 선언하지 않고,
유효성 검증해야 하는 부분이
내가 가진 잔고보다 사용할 금액이 크면 잔액부족인거니까
if (this.point (내가 가진 잔고) < amount( 사용할 금액) ) {
    잔고 부족 Exception
}

이렇게 정리될거고 명확한 표현일거라 말씀드렸습니다.

(this.point < amount ) == (this.point - amount < 0 ) == (currentPoint < minPoint(=0))
같은 표현입니다.